From e32541c26fa62df0b14c2ddb04e4937fbc7350b0 Mon Sep 17 00:00:00 2001
From: Joshua McCloskey <joshmccloskey@google.com>
Date: Tue, 31 May 2022 23:49:50 +0000
Subject: [PATCH] fpService#authWithPrompt uses correct user handle.

CTS > BYOD Managed Provisioning > Authentication Bound Keys

Verified Fingerprint-bound key test works as expected.

Test: Manually verified CTS
Bug: 231932206
Change-Id: I473c9c28cd0fbb01f4dd48447ddea8aa32834131
(cherry picked from commit f3650a6dee1ebe5f681699e4170c244e7bd7f9fc)
(cherry picked from commit c79346286e9cb03df8bf8417c15059aed0f0b41a)
Merged-In: I473c9c28cd0fbb01f4dd48447ddea8aa32834131
---
 .../biometrics/sensors/fingerprint/FingerprintService.java   | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/services/core/java/com/android/server/biometrics/sensors/fingerprint/FingerprintService.java b/services/core/java/com/android/server/biometrics/sensors/fingerprint/FingerprintService.java
index 5524291deeb3..bd42a17d5ec2 100644
--- a/services/core/java/com/android/server/biometrics/sensors/fingerprint/FingerprintService.java
+++ b/services/core/java/com/android/server/biometrics/sensors/fingerprint/FingerprintService.java
@@ -332,7 +332,7 @@ public class FingerprintService extends SystemService {
             if (!isKeyguard && !Utils.isSettings(getContext(), opPackageName)
                     && sensorProps != null && sensorProps.isAnyUdfpsType()) {
                 try {
-                    return authenticateWithPrompt(operationId, sensorProps, userId, receiver, opPackageName);
+                    return authenticateWithPrompt(operationId, sensorProps, callingUid, callingUserId, receiver, opPackageName);
 		} catch (PackageManager.NameNotFoundException e) {
 		    Slog.e(TAG, "Invalid package", e);
                 }
@@ -345,13 +345,14 @@ public class FingerprintService extends SystemService {
         private long authenticateWithPrompt(
                 final long operationId,
                 @NonNull final FingerprintSensorPropertiesInternal props,
+                final int uId,
                 final int userId,
                 final IFingerprintServiceReceiver receiver,
                 final String opPackageName) throws PackageManager.NameNotFoundException {
 
             final Context context = getUiContext();
             final Context promptContext = context.createPackageContextAsUser(
-                    opPackageName, 0 /* flags */, UserHandle.getUserHandleForUid(userId));
+                    opPackageName, 0 /* flags */, UserHandle.getUserHandleForUid(uId));
             final Executor executor = context.getMainExecutor();
 
             final BiometricPrompt biometricPrompt = new BiometricPrompt.Builder(promptContext)
-- 
2.17.1

